backend/core/src/listings/combined/combined-listing-query-builder.ts

```diff
@@ -0,0 +1,194 @@
+import { Listing } from "../entities/listing.entity"
+import { SelectQueryBuilder } from "typeorm"
+import { Pagination } from "nestjs-typeorm-paginate"
+import { CombinedListingTransformer } from "./combined-listing-transformer"
+import { ListingFilterParams } from "../dto/listing-filter-params"
+import { addFilters } from "../../shared/query-filter"
+import { combinedListingFilterTypeToFieldMap } from "./filter-type-to-field-map"
+import { OrderByFieldsEnum } from "../types/listing-orderby-enum"
+import { OrderParam } from "../../applications/types/order-param"
+import { HttpException, HttpStatus } from "@nestjs/common"
+import { CombinedListing } from "./combined-listing.entity"
+
+type OrderByConditionData = {
+  orderBy: string
+  orderDir: "DESC" | "ASC"
+  nulls?: "NULLS LAST" | "NULLS FIRST"
+}
+
+// REMOVE_WHEN_EXTERNAL_NOT_NEEDED
+export class CombinedListingsQueryBuilder extends SelectQueryBuilder<CombinedListing> {
+  limitValue?: number | "all"
+  pageValue?: number
+  addUnitSummaryFlag?: boolean
+
+  private static shouldPaginate(limit: number | "all", page: number) {
+    return limit !== "all" && limit > 0 && page > 0
+  }
+
+  private static getOrderByCondition(
+    orderBy: OrderByFieldsEnum,
+    orderDir: OrderParam
+  ): OrderByConditionData {
+    switch (orderBy) {
+      case OrderByFieldsEnum.mostRecentlyUpdated:
+        return { orderBy: "updated_at", orderDir }
+      case OrderByFieldsEnum.status:
+        return { orderBy: "status", orderDir }
+      case OrderByFieldsEnum.name:
+        return { orderBy: "name", orderDir }
+      case OrderByFieldsEnum.waitlistOpen:
+        return { orderBy: "is_waitlist_open", orderDir }
+      case OrderByFieldsEnum.unitsAvailable:
+        return { orderBy: "units_available", orderDir }
+      case OrderByFieldsEnum.mostRecentlyClosed:
+        return {
+          orderBy: "closed_at",
+          orderDir,
+          nulls: "NULLS LAST",
+        }
+      case OrderByFieldsEnum.mostRecentlyPublished:
+        return {
+          orderBy: "published_at",
+          orderDir,
+          nulls: "NULLS LAST",
+        }
+      // not sure what listings.marketingType refers to?
+      //case OrderByFieldsEnum.marketingType:
```

Comment:
marketing type is something that is currently only used in Detroit. They have "coming soon" properties and I'm guessing part of that code got carried over. So it can be deleted here

Reply:
Great!  Thanks for clearing that up